source/dxgi/dxgi_factory.cpp

```diff
@@ -127,20 +537,89 @@ HRESULT STDMETHODCALLTYPE DXGIFactory::GetWindowAssociation(HWND *pWindowHandle)
 }
 HRESULT STDMETHODCALLTYPE DXGIFactory::CreateSwapChain(IUnknown *pDevice, DXGI_SWAP_CHAIN_DESC *pDesc, IDXGISwapChain **ppSwapChain)
 {
+#ifndef RESHADE_IDXGIFACTORY_VTABLE
+	if (g_in_dxgi_runtime)
```

Comment:
Realized now that this check here should not be necessary, since these proxy objects should never be used within the DXGI runtime, it's really only needed for the vtable hooks.